test/net/sourceforge/kolmafia/textui/parsetree/FunctionTest.java

```diff
@@ -113,4 +121,578 @@ public static Stream<ScriptData> data() {
+        @Test
+        void int2_int() {
+          Function f =
+              makeFunction("f", DataTypes.VOID_TYPE, DataTypes.INT_TYPE, DataTypes.INT_TYPE);
+          makeValues(DataTypes.INT_TYPE);
+          assertFalse(f.paramsMatch(values, MatchType.EXACT, false));
+        }
+
+        // One int is required, two are provided
+        @Test
+        void int_int2() {
+          Function f = makeFunction("f", DataTypes.VOID_TYPE, DataTypes.INT_TYPE);
+          makeValues(DataTypes.INT_TYPE, DataTypes.INT_TYPE);
+          assertFalse(f.paramsMatch(values, MatchType.EXACT, false));
+        }
+
+        // int is required, typedef int is provided
+        @Test
+        void int_typedef() {
+          TypeDef td = new TypeDef("td", DataTypes.INT_TYPE, null);
+          Function f = makeFunction("f", DataTypes.VOID_TYPE, DataTypes.INT_TYPE);
+          makeValues(td);
+          assertFalse(f.paramsMatch(values, MatchType.EXACT, false));
+        }
+
+        // float is required, int is provided
+        @Test
+        void float_int() {
+          Function f = makeFunction("f", DataTypes.VOID_TYPE, DataTypes.FLOAT_TYPE);
+          makeValues(DataTypes.INT_TYPE);
+          assertFalse(f.paramsMatch(values, MatchType.EXACT, false));
+        }
+
+        // typedef int is required, int is provided
+        @Test
+        void typedef_int() {
+          TypeDef td = new TypeDef("td", DataTypes.INT_TYPE, null);
+          Function f = makeFunction("f", DataTypes.VOID_TYPE, td);
+          makeValues(DataTypes.INT_TYPE);
+          assertFalse(f.paramsMatch(values, MatchType.EXACT, false));
+        }
+
+        // typedef int is required, same typedef int is provided
+        @Test
+        void typedef_typedef() {
+          TypeDef td = new TypeDef("td", DataTypes.INT_TYPE, null);
+          Function f = makeFunction("f", DataTypes.VOID_TYPE, td);
+          makeValues(td);
+          assertTrue(f.paramsMatch(values, MatchType.EXACT, false));
+        }
+
+        // typedef int is required, different typedef int is provided
+        @Test
+        void typedef1_typedef2() {
+          TypeDef td1 = new TypeDef("td1", DataTypes.INT_TYPE, null);
+          TypeDef td2 = new TypeDef("td2", DataTypes.INT_TYPE, null);
+          Function f = makeFunction("f", DataTypes.VOID_TYPE, td1);
+          makeValues(td2);
+          assertFalse(f.paramsMatch(values, MatchType.EXACT, false));
+        }
+      }
+
+      @Nested
+      class VarArgs {
+        // The function does have a vararg parameter
+        // Tests for functions with valid varargs.
+
+        // vararg allowed, no matching args provided
+        @Test
+        void int_int0() {
+          Function f =
+              makeFunction(
+                  "f", DataTypes.VOID_TYPE, DataTypes.STRING_TYPE, vararg(DataTypes.INT_TYPE));
+          makeValues(DataTypes.STRING_TYPE);
+          // *** There is a varag, but we ran out of values before hitting it.
```

Comment:
`varag` -> `vararg`, and same for every time it occurs.

Reply:
I have corrected these comments, based on the next comment, but the mis-spelling is gone.